app/src/main/java/com/frb/books/asynctasks/PostRequestBooks.java

```diff
@@ -0,0 +1,72 @@
+package com.frb.books.asynctasks;
+
+import android.content.Context;
+import android.os.AsyncTask;
+import android.widget.Toast;
+
+import com.frb.books.Service.HttpService;
+import com.frb.books.classes.Error;
+import com.frb.books.classes.PostBookResponse;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.io.IOException;
+import java.net.URL;
+
+public class PostRequestBooks extends AsyncTask<String,Void,String>{
+    private HttpService httpService;
+    private String strRequest;
+    private Context appContext;
+    private String codResult;
+
+    public PostRequestBooks(String strRequest, Context appContext)
+    {
+        this.httpService = new HttpService();
+        this.strRequest = strRequest;
+        this.appContext = appContext;
+    }
+
+    @Override
+    protected void onPreExecute() {
+        httpService.setStrRequest(strRequest);
+    }
```

Comment:
E necesar set aici? Poate e mai bine sa fie mutat in constructor, la fel si set-urile din doInBackground